### PR TITLE
pythonPackage.uvicorn 0.11.5 -> 0.12.2 and fix tests on darwin

### PR DESCRIPTION
**Pull request by thanegill**
###### Motivation for this change
Update uvicorn and fix tests on darwin

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/uvicorn/default.nix [hunk 1]
```diff
@@ -7,16 +7,17 @@
...
-, pytest
+, pytestCheckHook
 , requests
 , isPy27
 }:
 
 buildPythonPackage rec {
   pname = "uvicorn";
-  version = "0.11.5";
+  version = "0.12.2";
```
Comment (mweinelt) [inline pkgs/development/python-modules/uvicorn/default.nix:17]: Version bump, but no hash change.

Comment (thanegill) [reply pkgs/development/python-modules/uvicorn/default.nix:20]: Oops. Fixed.

**Comment by wd15** [comment]
Built for Python 3.8, but not Python 3.7. Should it build for Python 3.7 and if not then add `isPy37` along with `isPy27?

```
$ nix-build . -A python37Packages.uvicorn
these derivations will be built:                                                                                                                             
  /nix/store/dsqapcd4nzxkasn4pw9psdqrhrlr10yl-python3.7-uvicorn-0.12.2.drv
...
ERROR: Could not find a version that satisfies the requirement typing-extensions; python_version < "3.8" (from uvicorn==0.12.2) (from versions: none)
ERROR: No matching distribution found for typing-extensions; python_version < "3.8" (from uvicorn==0.12.2)
builder for '/nix/store/dsqapcd4nzxkasn4pw9psdqrhrlr10yl-python3.7-uvicorn-0.12.2.drv' failed with exit code 1
error: build of '/nix/store/dsqapcd4nzxkasn4pw9psdqrhrlr10yl-python3.7-uvicorn-0.12.2.drv' failed
```

File: pkgs/development/python-modules/uvicorn/default.nix [hunk 2]
```diff
@@ -33,17 +37,22 @@ buildPythonPackage rec {
     wsproto
   ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/uvicorn/default.nix:38]:
```suggestion
  ] ++ lib.optional pythonOlder "3.8" typing-extensions;
```

**Comment by mweinelt** [comment]
This change breaks the httpx tests, they reproducibly get stuck at

```
============================= test session starts ==============================
platform linux -- Python 3.8.6, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
rootdir: /build/source, inifile: setup.cfg
plugins: asyncio-0.14.0, trio-0.6.0, cov-2.10.1
collected 584 items / 4 deselected / 580 selected

tests/test_api.py ..........                                             [  1%]
tests/test_asgi.py ..............                                        [  4%]
tests/test_auth.py ...                                                   [  4%]
```

**Comment by SuperSandro2000** [comment]
@thanegill can you fix the merge conflict and take a look at the hanging tests?

**Comment by mweinelt** [comment]
Superseded by #108602 . I've pulled in the darwin test fix.

**Pull request closed without merging**